CriticalMapsKit/Sources/ApiClient/Endpoint.swift

```diff
@@ -3,20 +3,34 @@ import Foundation
 /// A structure to define an endpoint on the Critical Maps API
 public struct Endpoint {
   public let baseUrl: String
-  public let path: String?
+  public let pathComponents: [String]
 
-  public init(baseUrl: String, path: String? = nil) {
+  public init(baseUrl: String, pathComponents: [String] = []) {
     self.baseUrl = baseUrl
-    self.path = path
+    self.pathComponents = pathComponents
+  }
+  
+  var url: String {
+    guard !pathComponents.isEmpty else {
+      return baseUrl
+    }
+    let path = pathComponents.joined(separator: "/")
+    return "\(baseUrl)/\(path)"
   }
 }
 
 public extension Endpoint {
+  static let locations = Self(baseUrl: cdnBaseUrl, pathComponents: ["locations"])
+  static let chatMessages = Self(baseUrl: cdnBaseUrl, pathComponents: ["messages"])
```

Comment:
the CDN caches the response for 60s. i recommend using the api-gateway instead.

Reply:
*just for messages though, locations should be cached via the CDN

Reply:
ah. good point!. Didn't pay attention on that one